sklearn/datasets/tests/test_openml.py

```diff
@@ -39,6 +39,10 @@
 test_offline = True
 
 
+# Do not use a cache for `fetch_openml` to avoid issues with `pytest-xdist`
```

Comment:
To me, the second part is even more worrying, even if the caching was concurrent-safe:

```suggestion
# Do not use a cache for `fetch_openml` to avoid concurrent writing
# issues with `pytest-xdist`.
# Furthermore sklearn/datasets/tests/data/openml/ is not always consistent
# with the version on openml.org. If one were to load the dataset outside of
# the tests, it may result in data that does not represent openml.org.
```